File: web-app/src/data/MockDataset.ts
export interface MockUser {
    name: string;
    role: string;
    serviceId: string;
    rankOrRelation: string;
    unitOrLink: string;
}

const officials: MockUser[] = [
    { name: "Gen. B. Rawat", role: "Officer", serviceId: "OFF-001", rankOrRelation: "General", unitOrLink: "Army HQ" },
    { name: "Col. S. Sharma", role: "Officer", serviceId: "OFF-002", rankOrRelation: "Colonel", unitOrLink: "1st Battalion" },
    { name: "Major R. Singh", role: "Officer", serviceId: "OFF-003", rankOrRelation: "Major", unitOrLink: "Northern Command" },
    { name: "Lt. K. Verma", role: "Officer", serviceId: "OFF-004", rankOrRelation: "Lieutenant", unitOrLink: "Signal Corps" },
    { name: "Capt. A. Khan", role: "Officer", serviceId: "OFF-005", rankOrRelation: "Captain", unitOrLink: "Intelligence Unit" },
    { name: "Cmdr. V. Iyer", role: "Officer", serviceId: "OFF-006", rankOrRelation: "Commander", unitOrLink: "Naval Base" },
    { name: "Brig. N. Gupta", role: "Officer", serviceId: "OFF-007", rankOrRelation: "Brigadier", unitOrLink: "Eastern Command" },
    { name: "Lt. Col. P. Reddy", role: "Officer", serviceId: "OFF-008", rankOrRelation: "Lt Colonel", unitOrLink: "Special Forces" },
    { name: "Major J. Das", role: "Officer", serviceId: "OFF-009", rankOrRelation: "Major", unitOrLink: "Artillery Reg" },
    { name: "Capt. M. Roy", role: "Officer", serviceId: "OFF-010", rankOrRelation: "Captain", unitOrLink: "Border Security" }
];

const soldiers: MockUser[] = [
    { name: "Hav. A. Patil", role: "Soldier", serviceId: "SOL-001", rankOrRelation: "Havildar", unitOrLink: "1st Inf Div" },
    { name: "Naik J. Singh", role: "Soldier", serviceId: "SOL-002", rankOrRelation: "Naik", unitOrLink: "2nd Inf Div" },
    { name: "Sep. R. Prasad", role: "Soldier", serviceId: "SOL-003", rankOrRelation: "Sepoy", unitOrLink: "3rd Inf Div" },
    { name: "L/Naik S. Kumar", role: "Soldier", serviceId: "SOL-004", rankOrRelation: "Lance Naik", unitOrLink: "4th Inf Div" },
    { name: "Hav. D. Joshi", role: "Soldier", serviceId: "SOL-005", rankOrRelation: "Havildar", unitOrLink: "5th Inf Div" },
    { name: "Sep. M. Ali", role: "Soldier", serviceId: "SOL-006", rankOrRelation: "Sepoy", unitOrLink: "6th Inf Div" },
    { name: "Naik P. Meena", role: "Soldier", serviceId: "SOL-007", rankOrRelation: "Naik", unitOrLink: "7th Inf Div" },
    { name: "Sep. G. Soren", role: "Soldier", serviceId: "SOL-008", rankOrRelation: "Sepoy", unitOrLink: "8th Inf Div" },
    { name: "L/Naik H. Gill", role: "Soldier", serviceId: "SOL-009", rankOrRelation: "Lance Naik", unitOrLink: "9th Inf Div" },
    { name: "Sep. K. Negi", role: "Soldier", serviceId: "SOL-010", rankOrRelation: "Sepoy", unitOrLink: "10th Inf Div" },
    { name: "Hav. L. Thapa", role: "Soldier", serviceId: "SOL-011", rankOrRelation: "Havildar", unitOrLink: "11th Inf Div" },
    { name: "Naik B. Gope", role: "Soldier", serviceId: "SOL-012", rankOrRelation: "Naik", unitOrLink: "12th Inf Div" },
    { name: "Sep. V. Murmu", role: "Soldier", serviceId: "SOL-013", rankOrRelation: "Sepoy", unitOrLink: "13th Inf Div" },
    { name: "L/Naik R. Rao", role: "Soldier", serviceId: "SOL-014", rankOrRelation: "Lance Naik", unitOrLink: "14th Inf Div" },
    { name: "Sep. T. Mishra", role: "Soldier", serviceId: "SOL-015", rankOrRelation: "Sepoy", unitOrLink: "15th Inf Div" },
    { name: "Hav. K. Yadav", role: "Soldier", serviceId: "SOL-016", rankOrRelation: "Havildar", unitOrLink: "16th Inf Div" },
    { name: "Naik S. Kurian", role: "Soldier", serviceId: "SOL-017", rankOrRelation: "Naik", unitOrLink: "17th Inf Div" },
    { name: "Sep. A. Pawar", role: "Soldier", serviceId: "SOL-018", rankOrRelation: "Sepoy", unitOrLink: "18th Inf Div" },
    { name: "L/Naik D. Bisht", role: "Soldier", serviceId: "SOL-019", rankOrRelation: "Lance Naik", unitOrLink: "19th Inf Div" },
    { name: "Sep. M. Dhillon", role: "Soldier", serviceId: "SOL-020", rankOrRelation: "Sepoy", unitOrLink: "20th Inf Div" }
];

const families: MockUser[] = [...officials, ...soldiers].map(user => ({
    name: `${user.name.split(" ").pop()} Family`,
    role: "Family",
    serviceId: `FAM-${user.serviceId}`,
    rankOrRelation: "Dependent",
    unitOrLink: user.serviceId
}));

export const MockDataset = {
    officials,
    soldiers,
    families,
    allUsers: [...officials, ...soldiers, ...families]
};

File: android-app/app/src/main/java/com/kavachsecurecomm/data/MockDataset.kt
package com.kavachsecurecomm.data

data class MockUser(
    val name: String,
    val role: String,
    val serviceId: String,
    val rankOrRelation: String,
    val unitOrLink: String
)

object MockDataset {
    val officials = listOf(
        MockUser("Gen. B. Rawat", "Officer", "OFF-001", "General", "Army HQ"),
        MockUser("Col. S. Sharma", "Officer", "OFF-002", "Colonel", "1st Battalion"),
        MockUser("Major R. Singh", "Officer", "OFF-003", "Major", "Northern Command"),
        MockUser("Lt. K. Verma", "Officer", "OFF-004", "Lieutenant", "Signal Corps"),
        MockUser("Capt. A. Khan", "Officer", "OFF-005", "Captain", "Intelligence Unit"),
        MockUser("Cmdr. V. Iyer", "Officer", "OFF-006", "Commander", "Naval Base"),
        MockUser("Brig. N. Gupta", "Officer", "OFF-007", "Brigadier", "Eastern Command"),
        MockUser("Lt. Col. P. Reddy", "Officer", "OFF-008", "Lt Colonel", "Special Forces"),
        MockUser("Major J. Das", "Officer", "OFF-009", "Major", "Artillery Reg"),
        MockUser("Capt. M. Roy", "Officer", "OFF-010", "Captain", "Border Security")
    )

    val soldiers = listOf(
        MockUser("Hav. A. Patil", "Soldier", "SOL-001", "Havildar", "1st Inf Div"),
        MockUser("Naik J. Singh", "Soldier", "SOL-002", "Naik", "2nd Inf Div"),
        MockUser("Sep. R. Prasad", "Soldier", "SOL-003", "Sepoy", "3rd Inf Div"),
        MockUser("L/Naik S. Kumar", "Soldier", "SOL-004", "Lance Naik", "4th Inf Div"),
        MockUser("Hav. D. Joshi", "Soldier", "SOL-005", "Havildar", "5th Inf Div"),
        MockUser("Sep. M. Ali", "Soldier", "SOL-006", "Sepoy", "6th Inf Div"),
        MockUser("Naik P. Meena", "Soldier", "SOL-007", "Naik", "7th Inf Div"),
        MockUser("Sep. G. Soren", "Soldier", "SOL-008", "Sepoy", "8th Inf Div"),
        MockUser("L/Naik H. Gill", "Soldier", "SOL-009", "Lance Naik", "9th Inf Div"),
        MockUser("Sep. K. Negi", "Soldier", "SOL-010", "Sepoy", "10th Inf Div"),
        MockUser("Hav. L. Thapa", "Soldier", "SOL-011", "Havildar", "11th Inf Div"),
        MockUser("Naik B. Gope", "Soldier", "SOL-012", "Naik", "12th Inf Div"),
        MockUser("Sep. V. Murmu", "Soldier", "SOL-013", "Sepoy", "13th Inf Div"),
        MockUser("L/Naik R. Rao", "Soldier", "SOL-014", "Lance Naik", "14th Inf Div"),
        MockUser("Sep. T. Mishra", "Soldier", "SOL-015", "Sepoy", "15th Inf Div"),
        MockUser("Hav. K. Yadav", "Soldier", "SOL-016", "Havildar", "16th Inf Div"),
        MockUser("Naik S. Kurian", "Soldier", "SOL-017", "Naik", "17th Inf Div"),
        MockUser("Sep. A. Pawar", "Soldier", "SOL-018", "Sepoy", "18th Inf Div"),
        MockUser("L/Naik D. Bisht", "Soldier", "SOL-019", "Lance Naik", "19th Inf Div"),
        MockUser("Sep. M. Dhillon", "Soldier", "SOL-020", "Sepoy", "20th Inf Div")
    )

    val families = (officials + soldiers).map { user ->
        MockUser(
            name = "${user.name.split(" ").last()} Family",
            role = "Family",
            serviceId = "FAM-${user.serviceId}",
            rankOrRelation = "Dependent",
            unitOrLink = user.serviceId
        )
    }

    val allUsers = officials + soldiers + families
}
